applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.cpp

```diff
@@ -0,0 +1,181 @@
+template<unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim,TNumNodes>::CalculateAndAddRHS(
+    VectorType& rRightHandSideVector,
+    NormalFluxVariables& rVariables )
+{
+    noalias(rVariables.TVector) = rVariables.NormalFlux * rVariables.Np * rVariables.IntegrationCoefficient;
+
+    GeoElementUtilities::
+        AssemblePBlockVector<0, TNumNodes>(rRightHandSideVector,
+                                                rVariables.TVector);
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<2, 2>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    const double dx_dxi = Jacobian(0, 0);
+    const double dy_dxi = Jacobian(1, 0);
+
+    const double ds = std::sqrt(dx_dxi * dx_dxi + dy_dxi * dy_dxi);
+
+    rIntegrationCoefficient = ds * Weight;
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<2, 3>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    const double dx_dxi = Jacobian(0, 0);
+    const double dy_dxi = Jacobian(1, 0);
+
+    const double ds = std::sqrt(dx_dxi * dx_dxi + dy_dxi * dy_dxi);
+
+    rIntegrationCoefficient = ds * Weight;
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<3, 3>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    double NormalVector[3];
+
+    NormalVector[0] = Jacobian(1, 0) * Jacobian(2, 1) - Jacobian(2, 0) * Jacobian(1, 1);
+
+    NormalVector[1] = Jacobian(2, 0) * Jacobian(0, 1) - Jacobian(0, 0) * Jacobian(2, 1);
+
+    NormalVector[2] = Jacobian(0, 0) * Jacobian(1, 1) - Jacobian(1, 0) * Jacobian(0, 1);
+
+    const double dA = std::sqrt(NormalVector[0] * NormalVector[0]
+        + NormalVector[1] * NormalVector[1]
+        + NormalVector[2] * NormalVector[2]);
+
+    rIntegrationCoefficient = dA * Weight;
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<3, 4>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    double NormalVector[3];
```

Comment:
I think you can call the other function instead of repeating the code, i.e.:

`this->CalculateIntegrationCoefficient<3,3>(rIntegrationCoefficient, Jacobian, Weight);`

Reply:
Same as above. It is not done. Do you have suggestion?